components/eam/src/physics/cam/cldera_passive_tracers.F90

```diff
@@ -0,0 +1,430 @@
+!===============================================================================
+! CLDERA passive tracers
+! provides dissipation rates and sources for diagnostic constituents
+!
+! AOA tracer implemented as defined in:
+! Gupta, A., Gerber, E.P. and Lauritzen, P.H. (2020) 
+! Numerical impacts on tracer transport: A proposed intercomparison test of Atmospheric 
+! General Circulation Models, Quarterly Journal of the Royal Meteorological Society,
+! doi:10.1002/qj.3881.
+!
+! E90 tracer implemented as defined in:
+! Abalos, M. et al. (2017)
+! Using the Artificial Tracer e90 to Examine Present and Future UTLS Tracer Transport in WACCM, 
+! Journal of the Atmospheric Sciences,
+! doi:10.1175/JAS-D-17-0135.1.
+! 
+! ST80_25 tracer implemented as defined in
+! Eyring, V. et al. (2013)
+! Overview of IGAC/SPARC Chemistry-Climate Model Initiative (CCMI) Community Simulations in 
+! Support of Upcoming Ozone and Climate Assessments
+!
+!===============================================================================
+
+module cldera_passive_tracers
+
+  use shr_kind_mod, only: r8 => shr_kind_r8
+  use spmd_utils,   only: masterproc
+  use ppgrid,       only: pcols, pver
+  use constituents, only: pcnst, cnst_add, cnst_name, cnst_longname
+  use cam_logfile,  only: iulog
+  use ref_pres,     only: pref_mid_norm
+
+  implicit none
+  private
+  save
+
+  ! Public interfaces
+  public :: cldera_passive_tracers_register         ! register constituents
+  public :: cldera_passive_tracers_implements_cnst  ! true if constituent is implemented by this package
+  public :: cldera_passive_tracers_init_cnst        ! initialize constituent field
+  public :: cldera_passive_tracers_init             ! initialize history fields, datasets
+  public :: cldera_passive_tracers_timestep_init    ! place to perform per timestep initialization
+  public :: cldera_passive_tracers_timestep_tend    ! calculate tendencies
+  public :: cldera_passive_tracers_readnl           ! read namelist options
+
+  ! ----- Private module data
+  integer, parameter :: ncnst=3  ! number of constituents implemented by this module
+
+  ! constituent names
+  character(len=8), parameter :: c_names(ncnst) = (/'AOA', 'E90j', 'ST80_25j'/)
```

Comment:
Question: do these automatically get added to available outputs? I do not see outfld calls for these, so I assume so.

Reply:
Yes, the `constituents` interface handles this after the advected species are registered with `cnst_add`. I have in the past output other intermediate quantities here for debugging, in which case I do include `addfld` and `outfld` calls for those fields.